yt_dlp/extractor/dailymotion.py

```diff
@@ -116,7 +116,7 @@ class DailymotionIE(DailymotionBaseInfoExtractor):
                     (?P<id>[^/?_&#]+)(?:[\w-]*\?playlist=(?P<playlist_id>x[0-9a-z]+))?
     '''
     IE_NAME = 'dailymotion'
-    _EMBED_REGEX = [r'<(?:(?:embed|iframe)[^>]+?src=|input[^>]+id=[\'"]dmcloudUrlEmissionSelect[\'"][^>]+value=)(["\'])(?P<url>(?:https?:)?//(?:www\.)?dailymotion\.com/(?:embed|swf)/video/.+?)\1']
+    _EMBED_REGEX = [rf'(?ix)<(?:(?:embed|iframe)[^>]+?src=|input[^>]+id=[\'"]dmcloudUrlEmissionSelect[\'"][^>]+value=)["\'](?P<url>{_VALID_URL[5:]})']
```

Comment:
`[5:]` is a bit obscure?

Maybe s/5/len('(?ix)')/ but constructing `_VALID_URL = f'(?ix){VALID_URL_TAIL}'` with the obvious definition of `_VALID_URL_TAIL` would be clearer. Or `re.sub(r'^\(\?[^)]*\)\s*', '', _VALID_URL)` would cater for any or no flag group (probably put that RE in utils as `FLAG_GROUP_RE`, say; and I see that Py3.11 really doesn't want whitespace before the group).

Reply:
With the two variables being right next to each other, I think it's fine. I don't want to overcomplicate the code.